### ISSUE TITLE
Compilation error with TBB

### ISSUE THREAD

**Issue by gdamiand** [CGAL/cgal#5381]
_Please use the following template to help us solving your issue._

## Issue Details

I have a compilation error with a code that use cgal concurrent compact container, with TBB.

```
In file included from /sps/liris/gdamiand/cgal/STL_Extension/include/CGAL/Uncertain.h:18:0,
                 from /sps/liris/gdamiand/cgal/STL_Extension/include/CGAL/assertions.h:350,
                 from /sps/liris/gdamiand/cgal/Profiling_tools/include/CGAL/Memory_sizer.h:16,
                 from /pbs/home/g/gdamiand/fast2darrangement/src/compute-arrangement-streamed.cpp:30:
/sps/liris/gdamiand/cgal/Profiling_tools/include/CGAL/Profile_counter.h:110:10: error: 'atomic' in namespace 'tbb' does not name a template type
     tbb::atomic<unsigned int> i;
```

I solved the error by adding 

```
#include <tbb/atomic.h>
```

into `Profiling_tools/include/CGAL/Profile_counter.h`  (line 64).

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux 64 bits
* Compiler: g++ (GCC) 10.2.0
* Release or debug mode: Release
* CGAL version: master
* Boost version: 1.64.0
* Other libraries versions if used (Eigen, TBB, etc.):  oneTBB 2021.1.1

**Comment by maxGimeno** [comment]
atomic should be taken from std. I'll make a PR to fix that.

**Comment by maxGimeno** [comment]
I am confused. In #5692 I replaced all occurences of atomic by std::atomic. How can you get that error in yesterday's testsuite ?

**Comment by gdamiand** [comment]
My bad; two versions of CGAL are mixed, one old version in /usr/local and master branch in my home directory.
Removing the local old version solves this issue.

By the way, is it normal that the two version were mixed ? I call cmake with -DCGAL_DIR=Mylocaldir ?

**Comment by maxGimeno** [comment]
It is possible that the default value for CGAL_DIR is taken in /usr/local, so if you don't specify it by hand at configuration time, you use examples from your home with the cgal from your /usr/local, or something like that.


Fixed by #5692.
